CONTRIBUTING.md

```diff
@@ -1,12 +1,32 @@
 # Contributing
 
+Firstly, thanks for taking the time to contribute! Here's hopefully some helpful
+info to get you started.
+
+## Before Adding a Feature
+
+Before a new feature or endpoint is added to this repo you'll want to make sure
+that there is agreement amongst the main parties using BSP. In order for this to
+happen please make sure there is an
+[Issue](https://github.com/build-server-protocol/build-server-protocol/issues)
+or a
+[Discussion](https://github.com/build-server-protocol/build-server-protocol/discussions/landing)
+started before you start working on your new feature. If you're just making a
```

Comment:
I feel it should be okay to submit a draft PR as a basis of discussion, with the expectation that it might not be accepted or heavily reworked before merging. Sometimes it's easier to just write / look at the code rather than laying out an abstract plan in detail.

Reply:
Good point. I went ahead and re-worked this section.